plugins/tableselection/plugin.js

```diff
@@ -1122,12 +1181,28 @@
 			CKEDITOR.document.appendStyleSheet( this.path + 'styles/tableselection.css' );
 		},
 
+		isElementIgnored: function( target ) {
```

Comment:
Why two of methods are located in `CKEDITOR.plugins.tableselection` and this one is inside `editor.plugins.tableselection`? I'd move them all to one place (preferably `CKEDITOR.plugins.tableselection`).

Reply:
The reason is keeping ignored elements as a part of an editor instance. Although, your idea about using custom data attribute instead of keeping reference into the element seems much better for me so we avoid an issue with accidental memory leaks and it handles other cases like already mentioned DOM unload.